fs/sync/sync.go

```diff
@@ -657,6 +657,15 @@ func (s *syncCopyMove) run() error {
 
 	s.startTrackRenames()
 
+	// If a max session duration has been defined, change the context to have a deadline
+	if fs.Config.MaxSession > 0 {
+		endTime := accounting.Stats.GetStartTime().Add(fs.Config.MaxSession)
+		fs.Infof(s.fdst, "Transfer session deadline: %s", endTime.Format("2006/01/02 15:04:05"))
+		newContext, cancel := context.WithDeadline(s.ctx, endTime)
+		s.ctx = newContext
+		s.cancel = cancel
```

Comment:
I don't think you want to overwrite the existing `s.ctx` or `s.cancel` as this context with deadline is just for March and its go-routines.